### PR TITLE
My homework for the first episode

### PR DESCRIPTION
**Pull request by telemachus**
Two things:
- I used MiniTest (since I'm somewhat familiar with it). I'll try MiniTest/Spec for next time maybe and work my way up to Rspec.
- I may have gone in a weird direction with the FoodBarge. Based on the name and the example you showed, I didn't want to reach into the animal's class and ask it what food it liked. It made sense to me that the barge should work sort of like a warehouse. But that got more complicated than I expected - maybe more complicated than it's worth.

Anyhow, looking forward to hearing your thoughts. Thanks.

### CODE REVIEW

File: food_barge.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+class FoodBarge
+  def initialize(stock = {})
+    @stock = Hash.new(0)
+    @stock.merge!(stock)
+  end
+
+  def count?(food)
+    @stock[food] || 0
```
Comment (jwo) [inline food_barge.rb:8]: Instead of `@stock[food] || 0`, I recommend replacing with

```
@stock.fetch(food, 0)
```

**Comment by jwo** [comment]
This looks great. I added a comment on a minor preference for accessing hashes, but otherwise looks good.

I think as long as one-thing-only knows what the animal will eat, it almost doesn't matter from a conceptual level. From a performance one, in an actual simulation you probably wouldn't want to loop through all food the barge has to find the cross-section of availability and compatibility. But: no one right answer here.

Note: I would make sure that in your code you can always execute

```
bundle exec rake
```

By convention, this will execute all tests. When I did that on yours, it would not pass the rspec because the zoo_spec was saying

```
require zoo
```

Of course, this is why most programs will choose 1 framework. I'm going to use both minitest and rspec throughout -- I think knowing both is beneficial.

**Comment by telemachus** [comment]
Thanks for the tip about `hash.fetch` - I never think of it, but I should. And sorry about the rake file. It's weird: when I run `bundle exec rake` it _does_ run the specs (though not the MiniTest tests). What's the error you get? I just want to check my set-up here.

(Btw, I won't usually be able to come to the same-day classes, but I should this Tuesday. Psyched about that.)

**Comment by jwo** [comment]
Actually, I can't push to your repository... here was the minor code change:

```
def food_for(list_of_acceptable_items)
  (@stock.keys - list_of_acceptable_items).sample
  #also get from stock
end
```

**Comment by telemachus** [comment]
Ah, no problem. I just checked out the branch from your repo. Thanks again for office hours.

**Pull request closed without merging**